client/my-sites/media-library/upload-url.jsx

```diff
@@ -49,11 +49,13 @@ export default React.createClass( {
 		}
 
 		MediaActions.clearValidationErrors( this.props.site.ID );
-		MediaActions.add( this.props.site.ID, this.state.value );
+		MediaActions.add( this.props.site.ID, this.state.value )
+			.then( () => {
```

Comment:
Do we need to react to the promise chain of the action? This really breaks the one-way data flow of Flux. We should be reacting to the change events emitted by the store itself, not the progress of a return value from the action creator. Were there complications in doing so?